core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
+   * of the exploration to Public/Private.
+   */
+  async expectExplorationToBePublish(): Promise<void> {
+    let publishButtonSelector = '.e2e-test-publish-exploration';
     if (this.isViewportAtMobileWidth()) {
-      await this.clickOn(mobileNavToggelbutton);
-      await this.clickOn(mobileSaveChangesButton);
+      publishButtonSelector = mobilePublishButton;
+      await this.clickOn(publishButtonDropDown);
+    }
+    const publishButton = await this.page.$(publishButtonSelector);
+    if (!publishButton) {
+      showMessage(
+        'Exploration is set to Public and is accessible to Oppia users.'
+      );
     } else {
-      await this.page.waitForSelector(`${saveChangesButton}:not([disabled])`);
-      await this.clickOn(saveChangesButton);
+      throw new Error(
+        'Exploration is set to Private and is not accessible to Oppia users.'
+      );
     }
+  }
 
-    await this.page.waitForFunction('document.readyState === "complete"');
-    await this.page.waitForSelector(saveDraftButton, {visible: true});
-    await this.clickOn(saveDraftButton);
-    await this.page.waitForSelector(saveDraftButton, {hidden: true});
-    await this.page.waitForFunction('document.readyState === "complete"');
+  /**
+   * This function add voice artists in the exploration.
+   */
+  async addVoiceArtists(voiceArtists: string[]): Promise<void> {
+    for (let i = 0; i < voiceArtists.length; i++) {
+      await this.clickOn(voiceArtistEditButton);
+      await this.clickOn(voiceArtistUserNameInputBox);
+      await this.page.waitForSelector(voiceArtistUserNameInputBox, {
+        visible: true,
+      });
+      await this.page.click(voiceArtistUserNameInputBox, {clickCount: 3});
+      await this.page.keyboard.press('Backspace');
+      await this.type(voiceArtistUserNameInputBox, voiceArtists[i]);
+      await this.clickOn(voiceArtistSaveButton);
+      await this.page.waitForSelector(
+        `div.e2e-test-voice-artist-${voiceArtists[i]}`,
+        {visible: true}
+      );
+
+      showMessage(voiceArtists[i] + ' has been added as a voice artist.');
+    }
+  }
+
+  /**
+   * This function helps to choose notification type by enabling/disabling
+   * the feedback toggle.
+   */
+  async optInToEmailNotifications(): Promise<void> {
+    await this.clickOn(feedbackToggle);
+    await this.expectEmailNotificationsToBeActivated();
+  }
+
+  /**
+   * Exception function to verify the choice of receiving feedback
+   * and suggestion notifications via email
+   */
+  async expectEmailNotificationsToBeActivated(): Promise<void> {
+    await this.page.waitForSelector('input[id="feedback-switch"]');
+    const input = await this.page.$('input[id="feedback-switch"]');
+
+    if (!input) {
+      throw new Error('Feedback switch input element not found.');
+    }
+    const feedbackSwitchIsActive = await input.evaluate(
+      input => (input as HTMLInputElement).checked
+    );
+
+    if (!feedbackSwitchIsActive) {
```

Comment:
Why are we saying that they're enabled if the feedback switch is *not* active?

Reply:
It's just a variable name. we are putting not(!) sign to show that it's not active.
Previously I have used isFeedbackSwitchActive/ isChecked you said it to change.

Reply:
Actually previously I was checking for feedback switch and assuming if feedback switch if OFF then suggestion email would be ON.(By default both are ON)

Now corrected to `suggestion switch` actually.